// @colbycommunications/eslint-config-colby/src/index.js
module.exports = {
    parser: 'babel-eslint',
    parserOptions: {
        ecmaVersion: 6,
        sourceType: 'module',
        ecmaFeatures: {
            jsx: true,
            modules: true,
            experimentalObjectRestSpread: true,
        },
    },
    extends: [
        'eslint:recommended',
        'airbnb',
        'prettier',
        'prettier/react',
        'plugin:jsx-a11y/recommended',
        'plugin:import/errors',
        'plugin:import/warnings',
        'plugin:jest/recommended',
        'plugin:react/recommended',
    ],
    rules: {
        'comma-dangle': [
            2,
            {
                arrays: 'always-multiline',
                objects: 'always-multiline',
                imports: 'always-multiline',
                exports: 'always-multiline',
                functions: 'ignore',
            },
        ],
        'react/jsx-uses-vars': 1,
        'react/jsx-indent': ['error', 4],
        'react/jsx-indent-props': ['error', 4],
        'react/display-name': 1,
        'react/destructuring-assignment': ['error', 'never'],
        'prefer-destructuring': ['error', { object: false, array: false }],
        'react/forbid-prop-types': ['error', { forbid: ['any'] }],
        quotes: ['error', 'single'],
        'jsx-quotes': ['error', 'prefer-double'],
        'no-unused-vars': 'warn',
        'no-console': ['error', { allow: ['warn', 'error', 'info'] }],
        'no-unexpected-multiline': 'warn',
        'arrow-body-style': ['error', 'as-needed', { requireReturnForObjectLiteral: true }],
        'react/prefer-stateless-function': 0,
        'import/extensions': 'ignorePackages',
        'react/jsx-filename-extension': [1, { extensions: ['.js', '.jsx'] }],
        'no-undef': 0,
        'no-plusplus': 0,
    },
};
